tox.ini

```diff
@@ -39,82 +40,63 @@ commands =
 description = Check code against coding style standards
 deps =
     black
+    codespell
     flake8
     flake8-docstrings
     flake8-copyright
     flake8-builtins
     pyproject-flake8
     pep8-naming
     isort
-    codespell
 commands =
+    codespell {[vars]lib_path}
+    codespell . --skip .git --skip .tox --skip build --skip lib --skip venv --skip .mypy_cache
     # pflake8 wrapper supports config from pyproject.toml
     pflake8 {[vars]all_path}
     isort --check-only --diff {[vars]all_path}
     black --check --diff {[vars]all_path}
-    codespell {[vars]lib_path}
-    codespell . --skip .git --skip .tox --skip build --skip lib --skip venv --skip .mypy_cache
 
-[testenv:static]
+[testenv:static-{charm,lib,unit,integration}]
 description = Run static analysis checks
+setenv =
+    unit: MYPYPATH = {[vars]tst_path}/unit
+    integration: MYPYPATH = {[vars]tst_path}/integration
 deps =
-    -r{toxinidir}/requirements.txt
     mypy
+    types-dataclasses
     types-PyYAML
-    pytest
-    pytest-operator
-    juju
-    types-setuptools
-    types-toml
     types-requests
+    charm: -r{toxinidir}/requirements.txt
+    lib: git+https://github.com/canonical/operator#egg=ops
+    unit: {[testenv:unit]deps}
+    integration: {[testenv:integration]deps}
 commands =
-    mypy {[vars]all_path} {posargs}
-    mypy --python-version 3.5 {[vars]lib_path} {posargs}
+    charm: mypy {[vars]src_path} {posargs}
+    lib: mypy --python-version 3.5 {[vars]lib_path} {posargs}
+    unit: mypy {[vars]tst_path}/unit {posargs}
+    integration: mypy {[vars]tst_path}/integration {posargs}
 
 [testenv:unit]
 description = Run unit tests
 deps =
-    -r{toxinidir}/requirements.txt
     pytest
     coverage[toml]
+    responses
+    -r{toxinidir}/requirements.txt
     deepdiff
 commands =
     coverage run \
       --source={[vars]src_path},{[vars]lib_path} \
-      -m pytest -v -s --tb native {posargs} {[vars]test_path}/unit
-    coverage report -m
+      -m pytest -v --tb native -s {posargs} {[vars]tst_path}/unit
+    coverage report
 
 [testenv:integration]
 description = Run integration tests
 deps =
+    aiohttp
     git+https://github.com/juju/python-libjuju.git
     #juju
     pytest
     git+https://github.com/charmed-kubernetes/pytest-operator.git
 commands =
-    pytest -v --tb native --log-cli-level=INFO -s {posargs} {toxinidir}/tests/integration
-
-[testenv:integration-lma]
```

Comment:
why are we dropping bundle tests?

Reply:
The bundle tests in this were more or less directly ripped from Prometheus, and don't even exercise Loki.

"Real" bundle integration tests will come later in a different repo which can tie everything together, as part of the pyramid testing strategy.